json-decode.go

```diff
@@ -170,7 +157,12 @@ func (cdc *Codec) decodeReflectJSONInterface(bz []byte, info *TypeInfo, rv refle
 	return nil
 }
 
+// CONTRACT: rv.CanAddr() is true.
```

Comment:
I guess we don't need these contracts spelled out, but I think it helps future devs who might want to try modifying the code.

Reply:
Great point, I believe we still need them. Those are guiding lights.